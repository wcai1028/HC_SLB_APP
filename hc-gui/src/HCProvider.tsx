import React from 'react'
import { Store } from 'redux'
import { A10Provider } from 'a10-gui-framework'

export interface IHCPRoviderProps {
  children: JSX.Element
  store?: Store<any>
  initState?: IObject
}

import appReducers from './redux/reducers'
import config from './settings/config'
const middlewares: any[] = []

const HCProvider = (props: IHCPRoviderProps) => {
  const { children, store, initState } = props
  const options: any = {
    initState,
    reducers: appReducers,
    middlewares,
    CONFIG: config,
    children,
  }
  if (store) {
    options.store = store
  }
  return <A10Provider {...options} />
}
export default HCProvider
