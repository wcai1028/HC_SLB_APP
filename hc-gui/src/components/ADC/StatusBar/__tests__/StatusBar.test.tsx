import * as React from 'react'
import * as ReactTestRenderer from 'react-test-renderer'
import * as moxios from 'moxios'
import HCProvider from 'src/HCProvider'

const renderTestComponent = () => {
  const StatusBar = require('../index').default
  return ReactTestRenderer.create(
    <HCProvider>
      <StatusBar />
    </HCProvider>,
  )
}
const renderTestComponentWithAllAttr = () => {
  const StatusBar = require('../index').default
  return ReactTestRenderer.create(
    <HCProvider>
      <StatusBar
        text="5"
        type="up"
        color="#00aa6c"
        isRect={true}
        isBig={true}
        tip={<React.Fragment />}
      />
      <StatusBar
        text="cp"
        type="down"
        color="#00aa6c"
        isRect={true}
        isBig={false}
        tip={<React.Fragment />}
      />
      <StatusBar
        text="pa"
        type="partial"
        color="#00aa6c"
        isRect={true}
        isBig={false}
        tip={<React.Fragment />}
      />
      <StatusBar
        text="dis"
        type="disable"
        color="#00aa6c"
        isRect={true}
        isBig={false}
        tip={<React.Fragment />}
      />
      <StatusBar
        text="none"
        type="none"
        color="#00aa6c"
        isRect={true}
        isBig={false}
        tip={<React.Fragment />}
      />
      <StatusBar
        text="un"
        type="unknow"
        color="#00aa6c"
        isRect={true}
        tip={<React.Fragment />}
      />
    </HCProvider>,
  )
}

const getComponentInstance = (testRenderer: any) => {
  const testInstance = testRenderer.root
  const testComponent = testInstance.find(
    (node: any) => node.type.name === 'StatusBar',
  )
  return testComponent.instance
}

beforeEach(() => {
  moxios.install()
})
afterEach(() => {
  moxios.uninstall()
})

describe('should render properly', () => {
  test('with EMPTY Attribute', (done: jest.DoneCallback) => {
    const testComponent = renderTestComponent()
    // wait all component are rendered ready
    setImmediate(() => {
      expect(testComponent.toJSON()).toMatchSnapshot()
      done()
    })
  })
  test('with ALL States of All Attributes', (done: jest.DoneCallback) => {
    const testComponent = renderTestComponentWithAllAttr()
    // wait all component are rendered ready
    setImmediate(() => {
      expect(testComponent.toJSON()).toMatchSnapshot()
      done()
    })
  })
})
