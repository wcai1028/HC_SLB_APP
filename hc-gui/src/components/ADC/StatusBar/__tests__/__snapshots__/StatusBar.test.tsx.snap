// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly with ALL States of All Attributes 1`] = `
Array [
  <div
    className="mock-a10-tooltip"
  >
    <span
      className="avatar avatarRectangleBig"
      style={
        Object {
          "backgroundColor": "#7ed321",
          "padding": "",
        }
      }
    >
      5
    </span>
  </div>,
  <div
    className="mock-a10-tooltip"
  >
    <span
      className="avatar avatarRectangleSmall"
      style={
        Object {
          "backgroundColor": "#ff2e48",
          "padding": "",
        }
      }
    >
      cp
    </span>
  </div>,
  <div
    className="mock-a10-tooltip"
  >
    <span
      className="avatar avatarRectangleSmall"
      style={
        Object {
          "backgroundColor": "#f5a623",
          "padding": "",
        }
      }
    >
      pa
    </span>
  </div>,
  <div
    className="mock-a10-tooltip"
  >
    <span
      className="avatar avatarRectangleSmall"
      style={
        Object {
          "backgroundColor": "#f1f8ff",
          "padding": "0 6px",
        }
      }
    >
      dis
    </span>
  </div>,
  <div
    className="mock-a10-tooltip"
  >
    <span
      className="avatar avatarRectangleSmall"
      style={
        Object {
          "backgroundColor": "#4d9fff",
          "padding": "0 6px",
        }
      }
    >
      none
    </span>
  </div>,
  <div
    className="mock-a10-tooltip"
  >
    <span
      className="avatar avatarRectangleSmall"
      style={
        Object {
          "backgroundColor": "#ffba47",
          "padding": "",
        }
      }
    >
      un
    </span>
  </div>,
]
`;

exports[`should render properly with EMPTY Attribute 1`] = `
<span
  className="avatar"
  style={
    Object {
      "backgroundColor": "",
      "padding": "",
    }
  }
>
  
</span>
`;
